R/utils.R

```diff
@@ -259,84 +259,333 @@ rowcol_to_dim <- function(row, col) {
+
+  if (lengt == 1 && !cond_acceptable_lengt_1) {
+    # Providing a single argument acceptable is only  `x`
+    sentence_unnamed <- ifelse(all_args_unnamed, "unnamed ", " ")
+    stop(
+      "Specifying a single", sentence_unnamed, "argument to `wb_dims()` is not supported.",
+      "\n",
+      "use any of `x`, `start_row` `start_col`. You can also use `rows` and `cols`, You can also use `dims = NULL`"
+    )
+  }
+  cnam_null <- is.null(args$col_names)
+  rnam_null <- is.null(args$row_names)
 
-  # wb_dims(rows, cols)
-  if (length(args) >= 2 && !exp_name) {
-    rows <- 1L
-    cols <- 2L
 
-    # wb_dims(rows = rows, cols = cols)
-    sel <- pmatch(nams, c("rows", "cols"))
-    valid <- length(sel[!is.na(sel)])
-    if (valid == 2) {
-      rows <- sel[rows]
-      cols <- sel[cols]
-    } else if (valid == 1) {
-      stop("found only one cols/rows argument")
+  if (!x_present) {
+    if (!cnam_null || !rnam_null) {
+      stop("`row_names`, and `col_names` should only be used if `x` is present.")
     }
+  }
+  rows_arg <- args$rows
+  #
+  x <- args$x
+  x_has_named_dims <- inherits(x, "data.frame") | inherits(x, "matrix")
+  if (x_has_named_dims && !is.null(rows_arg)) {
+    is_rows_a_colname <- rows_arg %in% colnames(x)
+
+    if (any(is_rows_a_colname)) {
+      stop("`rows` is the incorrect argument. Use `cols` instead.")
+    }
+  }
+  if (is.character(rows_arg)) {
+    warning("It's preferable to specify integers indices for `rows`", "See `col2int(rows)` to find the correct index.")
+  }
 
-    rows <- as.integer(args[[rows]])
-    cols <- col2int(args[[cols]])
+  rows_arg <- col2int(rows_arg, allow_null = TRUE)
+  cols_arg <- args$cols
+  x <- args$x
+  x_has_named_dims <- inherits(x, "data.frame") | inherits(x, "matrix")
 
-  } else {
+  # rows_and_cols_present <- all(c("rows", "cols") %in% nams)
 
-    if (cnam_null) col_names <- exp_name
-    if (rnam_null) row_names <- FALSE
 
-    assert_class(col_names, "logical")
-    assert_class(row_names, "logical")
+  # Find column location id if `cols` is named.
+  if (x_has_named_dims && !is.null(cols_arg)) {
+    is_cols_a_colname <- cols_arg %in% colnames(x)
 
-    # wb_dims(data.frame())
-    x <- as.data.frame(x)
-    rows <- seq_len(nrow(x) + col_names)
-    cols <- seq_len(ncol(x) + row_names)
+    if (any(is_cols_a_colname)) {
+      if (length(is_cols_a_colname) != 1) {
+        stop(
+          "Supplying multiple column names is not supported by the `wb_dims()` helper, use the `cols`  arguments instead.",
+          "\n Use a single `cols` at a time with `wb_dims()`"
+        )
+      }
+      col_name <- cols_arg
+
+      cols_arg <- which(colnames(x) == cols_arg)
+      message("Transforming col name = '", col_name, "' to `cols = ", cols_arg, "`")
+    }
+  }
 
+  if (!is.null(rows_arg)) {
+    assert_class(rows_arg, class = "integer", envir = parent.frame(n = 2), arg_nm = "rows_arg")
```

Comment:
n == 2 is two up? Like from the current to the parent to the grand parent?

Reply:
good catch. I tried putting this in a function at some point, but changed my mind and forgot to revert this change